src/spiral_stc.cpp

```diff
@@ -90,64 +101,193 @@ namespace full_coverage_path_planner
 
     return global_path;
   }
+
   std::list<gridNode_t> SpiralSTC::spiral(std::vector<std::vector<bool>> const &grid, std::list<gridNode_t> &init,
-                                          std::vector<std::vector<bool>> &visited)
+                                          double &yaw_start, std::vector<std::vector<bool>> &visited)
   {
-    int nRows = grid.size();
-    int nCols = grid[0].size();
-    // Spiral filling of the open space
-    // Copy incoming list to 'end'
-    std::list<gridNode_t> pathNodes(init);
-    // Create iterator for gridNode_t list and let it point to the last element of end
-    std::list<gridNode_t>::iterator it = --(pathNodes.end());
-    if (pathNodes.size() > 1) // if list is length 1, keep iterator at end
-      it--;                   // Let iterator point to second to last element
-
+    std::list<gridNode_t> pathNodes(init); // Copy incoming list to pathNodes
+    std::list<gridNode_t>::iterator it = --(pathNodes.end()); // Create iterator and let it point to the last element of end
+    if (pathNodes.size() > 1) // If list is length 1, keep iterator at end
+    {
+      it--; // Let iterator point to second to last element
+    }
     gridNode_t prev = *(it);
+
+    // Initialize spiral direction towards robot's y-axis
+    int dx = 0;
+    int dy = 1;
+    int dx_prev;
+    double yaw1 = yaw_start; // This value will change later on
```

Comment:
please use a more descriptive name

Reply:
I changed it (and others) in the latest commit.